addons/account/tests/test_account_journal.py

```diff
@@ -294,3 +294,27 @@ def test_alias_from_type(self):
                          'Journal alias owned by journal itself')
         self.assertEqual(journal_alias_2.alias_parent_thread_id, journal.id,
                          'Journal alias owned by journal itself')
+
+    def test_alias_journal_after_rename(self):
+        """ Test alias when journal is created with name matching existing alias """
+        company_name = 'company_1_data'
+        # create a journal and rename it
+        journal = self.env['account.journal'].create({
+            'name': 'Test journal',
+            'type': 'purchase',
+            'code': 'BILL1',
+        })
+        self.assertEqual(journal.alias_name, f'test-journal-{company_name}')
+        journal.name = 'test-journal-2'
+        # Create a new one with the same name as the first journal
+        journal_2 = self.env['account.journal'].create({
+            'name': 'Test journal',
+            'type': 'purchase',
+            'code': 'BILL2',
+        })
+        # check that the alias name is correctly set, name + company name + journal code
+        self.assertEqual(journal_2.alias_name,
+                        self.env['mail.alias']._sanitize_alias_name(f'test-journal-{company_name}-{journal_2.code}'))
+        self.assertTrue(journal_2.alias_id)
+        self.assertEqual(journal_2.alias_id.alias_parent_thread_id, journal_2.id,
+                        'Journal alias owned by journal itself')
```

Comment:
What exactly does this test ? I don't really understand, as your code seems to only improve the naming of the alias names